### ISSUE TITLE
[mesa 19.1.6]dEQP-VK.ssbo.layout.random.16bit.all_shared_buffer#40 failed due to timeout

### ISSUE THREAD

**Issue by renchenglei** [intel/external-mesa#129]
One dEQP-VK.ssbo.layout.* case failed due to "timeout" on Android 10 with mesa 19.1.6.
```
dEQP-VK.ssbo.layout.random.16bit.all_shared_buffer#40
```
**This issue also could be reproduced on Android 10 with mesa 19.0.6.
We have tried with upstream master branch(e4a52bd65), and this issue can't be reproduced. Need double confirm which changes to help fix this issue on upstream.**

Steps to reproduce this issue(run with cts packages):
```
run cts -m CtsDeqpTestCases -t dEQP-VK.ssbo.layout.random.16bit.all_shared_buffer#40
```

**Comment by renchenglei** [comment]
Same with https://github.com/intel/external-mesa/issues/131, this also could pass with tag 19.2.1

**Comment by renchenglei** [comment]
This issue could be fixed with 19.2.1. Have some bisec work, it is fixed with following 3 patches:
d27cc8f4968b7849cca99cb5200117240d76ddf2
3f9296b25ec6f716a6c4bc7b40f48eed11af65c2
191e1b1da4b1b6667f89768c5f8e86b8026754b7

**Comment by tpalli** [comment]
@renchenglei these do not seem valid hashes for Mesa master, which repository has those hashes?

**Comment by renchenglei** [comment]
@tpalli, yeah, so sorry, I update the new commit id on upstream 19.2.0 tag:
0b6da5bac66dc9abb9bd4617b156186c4fd3f92b
20bbc175a4eea5604357bae6690efa1bb1f37feb
f8bda81887219d9a56b5427c20be3e63b5c3d136

**Comment by tpalli** [comment]
@renchenglei Thanks! Yes, this makes sense as all of those commits are for optimizing the compiler performance so it makes these tests faster and they won't time out.
